plugin.video.newsy/default.py

```diff
@@ -0,0 +1,212 @@
+VID_TYPE  = ['M3U8','MP4'][int(REAL_SETTINGS.getSetting('Video_Type'))]
+
+def log(msg, level=xbmc.LOGDEBUG):
+    if DEBUG == True:
+        if level == xbmc.LOGERROR:
+            msg += ' ,' + traceback.format_exc()
+        xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + (msg), level)
+   
+def getParams():
+    param=[]
+    if len(sys.argv[2])>=2:
+        params=sys.argv[2]
+        cleanedparams=params.replace('?','')
+        if (params[len(params)-1]=='/'):
+            params=params[0:len(params)-2]
+        pairsofparams=cleanedparams.split('&')
+        param={}
+        for i in range(len(pairsofparams)):
+            splitparams={}
+            splitparams=pairsofparams[i].split('=')
+            if (len(splitparams))==2:
+                param[splitparams[0]]=splitparams[1]
+    return param
+                 
+socket.setdefaulttimeout(TIMEOUT)
+class Newsy():
+    def __init__(self):
+        log('__init__')
+        self.cache = SimpleCache()
+
+            
+    def openURL(self, url):
+        log('openURL, url = ' + str(url))
+        try:
+            cacheResponse = self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+            if not cacheResponse:
+                request = urllib2.Request(url)
+                request.add_header('User-Agent','Mozilla/5.0 (Windows; U; MSIE 9.0; Windows NT 9.0; en-US)')
+                response = urllib2.urlopen(request, timeout=TIMEOUT).read()
+                self.cache.set(ADDON_NAME + '.openURL, url = %s'%url, response, expiration=datetime.timedelta(hours=6))
+            return self.cache.get(ADDON_NAME + '.openURL, url = %s'%url)
+        except urllib2.URLError, e:
+            log("openURL Failed! " + str(e), xbmc.LOGERROR)
+        except socket.timeout, e:
+            log("openURL Failed! " + str(e), xbmc.LOGERROR)
+        except Exception, e:
+            log("openURL Failed! " + str(e), xbmc.LOGERROR)
+            xbmcgui.Dialog().notification(ADDON_NAME, LANGUAGE(30001), ICON, 4000)
+            return ''
+             
+
+    def buildMainMenu(self):
+        self.addLink('Live'     ,'' ,0)
+        self.addDir('Browse'    ,'0',1)
+        
+        
+    def buildLiveLink(self):
+        soup = BeautifulSoup(self.openURL(LIVEURL), "html.parser")
+        link = 'http:' + soup('div' , {'class': 'live-player'})[0].find('script').get('src')
+        path = '{'+self.openURL(link).split('}; // end config')[0].split('var jwConfig = {')[1]+'}'
+        item = json.loads(path)
+        self.playVideo('Newsy Live',item['playlist'][0]['sources'][0]['file'])
+            
+                         
+    def pagination(self, seq, rowlen):
+        for start in xrange(0, len(seq), rowlen):
+            yield seq[start:start+rowlen]
+
+            
+    def buildRSS(self, start=0, end=24):
+        data    = feedparser.parse(RSSURL)['entries']
+        data    = list(self.pagination(data, end))
+        start   = 0 if start >= len(data) else start
+        for item in data[start]:
```

Comment:
Sure... I ran some tests in the past and didn't observe compile times differences since the scripts are relatively small. Nevertheless I can start incorporating a entry point.